### ISSUE TITLE
Smart Blueprint: Component Interface Design

### ISSUE THREAD

**Issue by simanadler** [mesh-for-data/mesh-for-data#612]
Define the structures for the input and output of the main components:

Data Governance Interface - input same as today, any changes required in existing output? Related to taxonomy work which is in progress

Non-Governance Policy Interface 

Optimizer - requires input of previous 2 components, and its output should be everything required to generate the blueprint

Blueprint generator - input TBD, output = new blueprint structure

Parent Issues: #93 

Dependencies: #594 #593 #595

**Comment by shlomitk1** [comment]
Two comments regarding Governance policies:
1. A higher abstraction of geographies is required.  
2. Current implementation of finding a location for writing data is sub-optimal. Instead of iterating over all available destinations and sending queries for each one of them, it would be better to have a single query: what are possible locations and what actions need to be taken?

**Comment by simanadler** [comment]
> Two comments regarding Governance policies:
> 
>     1. A higher abstraction of geographies is required.
> 
@shlomitk1 I found the following open source package.  Do you think this would be helpful? https://github.com/dr5hn/countries-states-cities-database
>     2. Current implementation of finding a location for writing data is sub-optimal. Instead of iterating over all available destinations and sending queries for each one of them, it would be better to have a single query: what are possible locations and what actions need to be taken?
@shlomitk1 While I agree this isn't optimal, we need to do something that is generic enough to be used by different policy managers.  Perhaps such optimization could be done in the policy manager connector though?

**Comment by shlomitk1** [comment]
@simanadler 
1. What I meant is that it is not reasonable to have (let's say we have 250 countries in the taxonomy, according to the database you sent) 250x250 governance rules about whether data can be moved between locations. By higher abstractions I meant grouping geographies in a certain way.
2. Agree to hide the complexity in the connector. From manager perspective it should be one query.

**Comment by simanadler** [comment]
> @simanadler
> 
>     1. What I meant is that it is not reasonable to have (let's say we have 250 countries in the taxonomy, according to the database you sent) 250x250 governance rules about whether data can be moved between locations. By higher abstractions I meant grouping geographies in a certain way.
> 
@shlomitk1 Yes, I agree with you completely.  Rather than creating a rule per country, rules should be defined either based on a position in a hierarchy or according to a grouping.  The package I found only has a very simple hierarchy (country/state/city) and can't support for example EU.  It also doesn't have a mechanism for grouping countries.  In a previous project I worked on we implemented all of this but I'm still looking for an open source equivalent.

>     2. Agree to hide the complexity in the connector. From manager perspective it should be one query.
